lib/protocol/protocol.go

```diff
@@ -813,41 +808,38 @@ func (c *rawConnection) shouldCompressMessage(msg message) bool {
 // BEP message is sent before terminating the actual connection. The error
 // argument specifies the reason for closing the connection.
 func (c *rawConnection) Close(err error) {
-	c.once.Do(func() {
+	c.sendCloseOnce.Do(func() {
 		done := make(chan struct{})
-		c.sendClose <- asyncMessage{&Close{err.Error()}, done}
-		<-done
-
-		// No more sends are necessary, therefore closing the underlying
-		// connection can happen at the same time as the internal cleanup.
-		// And this prevents a potential deadlock due to calling c.receiver.Closed
-		go c.commonClose(err)
+		c.send(&Close{err.Error()}, done)
+		select {
+		case <-done:
+		case <-c.closed:
+		}
 	})
+
+	// No more sends are necessary, therefore closing the underlying
+	// connection can happen at the same time as the internal cleanup.
+	// And this prevents a potential deadlock due to calling c.receiver.Closed
+	go c.internalClose(err)
```

Comment:
This comment is kind of funky now given there is only two closes rather than 3? Or am I just misunderstanding what it means.

Reply:
The comment should not depend on how many closes there is. Does this change make it clearer?